ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/transaction/arjunacore/jca/TransactionImporterImple.java

```diff
@@ -230,7 +230,12 @@ public Set<Xid> getInflightXids(String parentNodeName) {
 	 * @param timeout
 	 * @return
 	 */
-	private TransactionImportResult addImportedTransaction(TransactionImple recoveredTransaction, Xid mapKey, Xid xid, int timeout) {
+	private TransactionImportResult addImportedTransaction(TransactionImple recoveredTransaction, Xid mapKey, Xid xid, int timeout) throws XAException {
+		if (!TxControl.isEnabled()) {
+			// TODO needs looking at for the right code and so on and maybe it's different for recover than it is when importing a new transaction
+			throw new XAException("Transaction system is shutting down");
```

Comment:
Please also cite a JBTM for resolving this TODO here too and use a blocking JBTM related to https://issues.redhat.com/browse/JBTM-3893

Reply:
(probably the same blocking JBTM as my other comment about a similar file though :) )

Reply:
Done. Thanks